soufi/finders/alpine.py

```diff
@@ -139,14 +139,18 @@ def _parse_apkbuild(self, path: Path) -> dict:
         # package is handled in a function in the APKBUILD. We can strip that
         # out for the purposes of matching subpackages.
         subpackages = [item.split(':', 1)[0] for item in parsed[1].split()]
-        sha512sums = lambda x: dict(zip(x[1::2], x[::2]))  # noqa: E731
+        # SHA-512 checksums for source archives are provided as a big flat list
+        # of the form (DIGEST, NAME, [...])  Turn it into a mapping of
+        # name->digest for validation lookups later.
+        sha512sums = parsed[5].split()
+        sha512sums_map = dict(zip(sha512sums[1::2], sha512sums[::2]))
```

Comment:
The problem I have is that there's "magic" numbers involved. Why every second item of the sequence, for example? Sometimes it's better to assign more lvalues to make the code more readable, which is also why I was intimating that it should be in a separate function to make it easier to do that.

Reply:
Oh god damn it Github, why can't you treat my two accounts as one and the same like I told you to.

Reply:
I... What? Those aren't magic numbers, it's just Python slice notation.

It's all explained in the comment, but I guess binding those slices to lvalues first can't hurt anything.

Reply:
They're magic in the sense that as a casual reader I have no insight into what they relate to. Unless you're intimately familiar with the format of the sha512sums string, it's hard to see what the code is doing. The comment didn't help as you've transcribed the actual string into a python-style format in the comment. So while it's a beautiful piece of code, it's hard to read for a newbie and understand what's going on at first glance.